.github/workflows/precommit.yaml

```diff
@@ -21,7 +21,5 @@ jobs:
         run: pip install -r requirements.lock
       - name: Run pre-commit
         run: pre-commit run --all-files
-      - name: Run gitlint on the latest commit message
-        run: >-
-          pre-commit run gitlint --hook-stage commit-msg --commit-msg-filename
-          .git/COMMIT_EDITMSG
+      - name: Run gitlint on CI with pre-commit
+        run: pre-commit run --hook-stage manual gitlint-ci
```

Comment:
Which commits does it check? Latest? entire log? PR content?

Reply:
latest commit by default. can be configured for more, but I don't think we want that.

Reply:
Suppose that a PR has more than one commit (happens from time to time) then we're not covered.
Also, I see for the current PR it complains over some sort of merge commit, which is not the latest commit.

I think it should test the commits between the target branch and the PR branch. I don't know how the command should look like exactly here, but the I think you get those git references "for free" inside a github action.

Reply:
I'm not seeing the added value for implementing this under precommit rather than a direct CI test (action or pipeline).
I feel like we're having a harder time running gitlint the way we want, with it encapsulated as a pre-commit hook.

We already have it solved in repos under redhat-appstudio both as workflow and as pipeline. Are we gaining anything with the current approach?

Reply:
Implementing it under precommit will allow having only one source for gitlint configuration, e.g: the version of gitlint we are using for both local and CI tests.
However I'm not against running it as a separate workflow, I'll check if the current issue can be easily solved